src/mps/projmposum.jl

```diff
@@ -128,10 +105,60 @@ ProjMPOSum `P`, and `ortho` is a String which can take
 the values `"left"` or `"right"` depending on the
 sweeping direction of the DMRG calculation.
 """
-function noiseterm(P::ProjMPOSum, phi::ITensor, dir::String)
-  nt = noiseterm(P.pm[1], phi, dir)
-  for n in 2:length(P.pm)
-    nt += noiseterm(P.pm[n], phi, dir)
+function noiseterm(A::AbstractSum, phi::ITensor, dir::String)
+  nt = noiseterm(terms(A)[1], phi, dir)
+  for n in 2:length(terms(A))
+    nt += noiseterm(terms(A)[n], phi, dir)
   end
   return nt
 end
+
+"""
+    disk(ps::AbstractSum; kwargs...)
+
+Call `disk` on each term of an AbstractSum, to enable
+saving of cached data to hard disk.
+"""
+function disk(sum::AbstractSum; disk_kwargs...)
+  return set_terms(sum, [disk(term; disk_kwargs...) for term in terms(sum)])
```

Comment:
To follow the style I suggest above, I would change this to use `map`:
```julia
set_terms(sum, map(term -> disk(term; disk_kwargs...), terms(sum)))
```
It's slightly more general since `map` could output something besides a `Vector`, and will try to stick to the type returned by `terms`.

Reply:
Agreed, and less verbose too